modules/geolocation/geolocation.py

```diff
@@ -129,3 +129,35 @@ def calculate_pixel_to_geo_mapping(self):
 
         # Return matrix product of mappedGeoMatrix and sourcePixelMatrixInverse
         return (mappedGeoMatrix.dot(sourcePixelMatrixInverse))
+
+    def map_location_from_pixel(self, transformation_matrix, pixels):
+        """
+        Maps Geographical Location Coordinates in the destination image
+        
+        Parameters
+        -------
+            transformation_matrix : np.array(shape=(3,3))
+            pixels : np.array(shape=(5,2))
+
+        Returns
+        -------
+            np.array(shape=(5,2))
+        """
+
+        # Express all 2D coordinates of pixels as 3D coordinates with z value = 1
+        pixels = np.insert(pixels, 2, 1, axis = 1)
```

Comment:
Few things I'm not following on: 1) How come z=1? Are we not assuming that z=0 making the plane perpendicular to the horizon?

Reply:
z=1 as the transformation is in R^3, but the input and output are in R^2. So it is required to convert the input to R^3 (this is called homogenization). After the transform, the result is also R^3 but needs to be converted to R^2 (dehomogenization). The reason this is done is so that we can use a linear (homogeneous) transformation (matrix multiplication) with a higher dimension instead of involving a matrix multiplication followed by an addition (a non-linear transformation).

See https://homepages.inf.ed.ac.uk/rbf/CVonline/LOCAL_COPIES/BEARDSLEY/node3.html (b) for an explanation (there was a better one than this for covering homogenization but I can't find it unfortunately).

Reply:
Ohhhhhhhhhhhhhhh right, my apologies, this was a really uninformed question lol I thought z=0 was from our plane thing. This is math 115 stuff I should've known, my apologies